src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -0,0 +1,341 @@
+        {
+            try
+            {
+                await _project.RemoveLegacyCSProjPackageAsync(packageIdentity.Id);
+            }
+            catch (Exception e)
+            {
+                nuGetProjectContext.Log(MessageLevel.Warning, e.Message, packageIdentity, _project.Name);
+                return false;
+            }
+
+            return true;
+        }
+
+        #endregion
+
+        private static string[] GetProjectReferences(PackageSpec packageSpec)
+        {
+            // There is only one target framework for legacy csproj projects
+            var targetFramework = packageSpec.TargetFrameworks.FirstOrDefault();
+            if (targetFramework == null)
+            {
+                return new string[] { };
+            }
+
+            return targetFramework.Dependencies
+                .Where(d => d.LibraryRange.TypeConstraint == LibraryDependencyTarget.ExternalProject)
+                .Select(d => d.LibraryRange.Name)
+                .Distinct(StringComparer.OrdinalIgnoreCase)
+                .ToArray();
+        }
+
+        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        {
+            var frameworkSorter = new NuGetFrameworkSorter();
+
+            return packageSpec
+                .TargetFrameworks
+                .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName))
+                .GroupBy(p => p.PackageIdentity)
+                .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
+                .ToArray();
+        }
+
+        private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework)
+        {
+            return libraries
+                .Where(l => l.LibraryRange.TypeConstraint == LibraryDependencyTarget.Package)
+                .Select(l => ToPackageReference(l, targetFramework));
+        }
+
+        private static PackageReference ToPackageReference(LibraryDependency library, NuGetFramework targetFramework)
+        {
+            var identity = new PackageIdentity(
+                library.LibraryRange.Name,
+                library.LibraryRange.VersionRange.MinVersion);
+
+            return new PackageReference(identity, targetFramework);
+        }
+
+        private async Task<PackageSpec> GetPackageSpec()
+        {
+            var projectReferences = _project.GetLegacyCSProjProjectReferencesAsync(_desiredPackageReferenceMetadata)
+                .Result
+                .Select(ToProjectRestoreReference);
+
+            var packageReferences = _project.GetLegacyCSProjPackageReferencesAsync(_desiredPackageReferenceMetadata)
+                .Result
+                .Select(ToPackageLibraryDependency);
+
+            var projectTfi = new TargetFrameworkInformation()
+            {
+                FrameworkName = await _project.GetTargetNuGetFramework(),
```

Comment:
>GetTargetNuGetFramework [](start = 47, length = 23)

Is this a method we call directly on DTE project? We should minimize logic in a wrapper type.
also, method name should end in Async